File: backend/src/Domain/Models/QuestionOption.cs
namespace backend.src.Domain.Models;

public class QuestionOption
{
    public int QuestionOptionId { get; set; }            // PK
    public int QuestionId { get; set; }          // FK -> Question
    public string FallbackText { get; set; }
    public string OptionValue { get; set; } = "";
    public int DisplayOrder { get; set; }

    public Question Question { get; set; } = null!;
    public ICollection<OptionText> Texts { get; set; } = new List<OptionText>();
}

File: frontend/src/types/QuestionOption.ts
export type CreateQuestionOptionDto = {
  questionId: number;
  fallbackText: string;
  optionValue: string;
  displayOrder: number;
};

export type QuestionOptionDto = {
  questionOptionId: number;
  questionId: number;
  fallbackText: string;
  optionValue: string;
  displayOrder: number;
};
